### DISCUSSION TITLE
Multiple html-button-response

### DISCUSSION THREAD

**Discussion by GaiaCo** [jspsych/jsPsych#1185] [Q&A]
Hello, 
I am trying to display two images with two response buttons for each image, one parallel to the other. One on the left side of the screen and one on the right. 
For the moment I am doing so by adding everything in the stimulus part of an html-keyboard-response (code attached) and taking the data from a JSON file, but since I would like to change both the images and the alternatives to answer in each trial, this can become a bit of a problem (especially when I have to calculate accuracy for feedback...). 
For this reason I wanted to modify the html-button-response plugin in order to vary the position of the first image with buttons and in order to add a second image with buttons. I am not really sure where to start though since I just started coding in JS and 
Could this make sense or should I just do with the method I am using? Is there another method I did not think about? 

Thank you for your answers :)

This is the html code I am currently using in an html-keyboard-response type. 

```js
stimulus: function() { //for now only Block 1: difficult on the left vs. difficult on the right, easy on the left vs. easy on the right
                        var html="<img src='"+jsPsych.timelineVariable('ImageL', true)+"'>"; html += "<font color = 'white'>_______________________________________________________________</font>";
                        html +="<img src='"+jsPsych.timelineVariable('ImageR', true)+"'>"; html += jsPsych.timelineVariable('TextL', true); html += "<font color = 'white'>__________________________</font>";
                        html += jsPsych.timelineVariable('TextR', true);
                        return html;
                    },
                    prompt: "<h1>Please categorize the color on the side you think the dots were moving.</h1>",
                    choices: ['z', 'x', 'n', 'm'],
                    response_ends_trial: true,
```

**Comment by jodeleeuw** [comment]
Hi @GaiaCo ,

If you end up modifying the plugin, you'll essentially be doing the same thing that you are currently doing with the `stimulus` parameter, but shifting that code into the plugin. That does make sense to do if it'll make it a lot easier to pass parameters for your trials.

Another idea would be to write a separate function that you can pass the images and button labels into, and it generates the corresponding HTML. This might make your trial code cleaner. 

Finally, instead of using a hidden line to push the images to one side of the screen, you can use CSS rules. For example, this is one way to display two images side by side:

```js
stimulus: '<div style="position: relative; width: 600px; height: 400px;">'+
	'<img src="img1.png" style="position: absolute; top: 0px; left: 0px;">'+
	'<img src="img2.png" style="position: absolute; top: 0px; right: 0px;">'+
	'</div>
```
